### DISCUSSION TITLE
Need help creating a Teslamate backup

### DISCUSSION THREAD

**Discussion by gshanson** [teslamate-org/teslamate#3551] [Q&A] [answered]
I need some help is creating a backup of my Teslamate.
I followed the instructions provided in docs.teslamate.org/docs/maintenance/backup_restore/.
The command I used was "docker compose exec -T database pg_dump -U teslamate teslamate > /backuplocation/teslamate.bck"
The response was "-bash:  /backupolocation/teslamate.bck: No such file or directory"
I am not very knowledgeable about these commands. 
Can someone tell me what I'm doing wrong?
Here is a screenshot of what I got.
![image](https://github.com/teslamate-org/teslamate/assets/58371763/4e4a73d3-234e-415e-9622-49afa0f5ce3c)

**Comment by cwanja** [comment]
Does the directory `/backuplocation` exist? If you are in your users home directory, just drop the file there and move it later.

**Reply by gshanson** [reply]
No, I don't think it exists.
So i tried to create the dir and I got this response:
![image](https://github.com/teslamate-org/teslamate/assets/58371763/6431c97c-2e1b-4a1b-b23e-c18de9044451)

**Reply by DrMichael** [reply]
Get rid of the leading "/"...

**Reply by gshanson** [reply]
Ok, so I made the backuplocation directory and then the backup command, it still didn't work.
You said to "drop the file there and move it later" 
I don't understand, I don't have the teslamate.bck file yet.  
![image](https://github.com/teslamate-org/teslamate/assets/58371763/e34093ec-ac1a-4830-87d4-ae0d649fc662)

**Reply by cwanja** [reply, marked as answer]
You are still using the `/backuplocation` in your command. As @DrMichael mentioned, drop the leading `/` from your `docker compose` command. 

Try this:
`docker compose exec -T database pg_dump -U teslamate teslamate > backuplocation/teslamate.bck`

**Reply by gshanson** [reply]
That worked!!!
Thank you all for your help.
